### DISCUSSION TITLE
128x128 ST7735 display with alignment issues

### DISCUSSION THREAD

**Discussion by shan-mcarthur** [Bodmer/TFT_eSPI#2573] [Compatible displays and setup files]
I have a [HiLetGo 128x128 ST7735 display](https://www.amazon.com/dp/B073R6SQRY?psc=1&ref=ppx_yo2ov_dt_b_product_details) that I am having alignment issues with and wonder what settings I should be looking at.  My board is a SparkFun ESP32-WROOM Thing Plus C.  The labels on the display were incredibly confusing (because they use I2Cish terms), but I have them wired to SPI like this:   

Display   ESP32
--------------------
LED  -> 3v3
SCK  -> SCK (GPIO18)
SDA -> MOSI (SPIO23)
AO -> GPIO14
Reset -> 3v3
CS -> GPIO22
GND -> GND
VCC -> 3v3

My sketch just prints Hello World after initializing the display.
`  tft.init();
  tft.setRotation(1);
  tft.fillScreen(TFT_BLACK);
  tft.setCursor(0,0,2);
  tft.setTextColor(TFT_WHITE);
  tft.println ("Hello World!");
`
If I define the screen as 128x128, the H is clipped off the left side of the screen and there is a white line down the right side and there is a line of random colors on the bottom of the screen.

![20230409_165133](https://user-images.githubusercontent.com/126544266/230802640-92b86fd1-d939-40c5-b80a-b840e283779e.jpg)

If I define the screen as 128x160, the bottom noise goes away, but the text clipping and white line down the right are still there.

Thoughts?

**Comment by shan-mcarthur** [comment]
I think I found the issue, although I don't understand the concept at all.  I located another forum and someone mentioned using greentab.  My screen came with a red tab, but I noticed some documentation in some of the setup files that encouraged me to experiment with each of them.  Sure enough, when I used ST7735_GREENTAB, my issues went away.

What is the meaning of GREENTAB, REDTAB, BLUETAB, and the various subvarients of those macros?  I don't see anything specifically in this library so it is likely an underlying dependency, but it is important for the configuration here to work.  Is there a basic concept I should understan with these macros?  The biggest challenge is that I typically rip the protective film off the display right away and dispose of them.  That colored label gets thrown out and there doesn't seem to be anything on the board or the documentation that would give me the proper indication...

**Reply by Bodmer** [reply]
See comments  here:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setup.h#L94

**Comment by uweschaller** [comment]
old post, but still helped a lot. mine was set to **ST7735_GREENTAB** but only works with **ST7735_REDTAB**
